src/sync/app_credentials.cpp

```diff
@@ -0,0 +1,116 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or utilied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "app_credentials.hpp"
+#include "../external/json/json.hpp"
+
+namespace realm {
+namespace app {
+
+std::string const kAppProviderKey = "provider";
+
+IdentityProvider const IdentityProviderAnonymous = "anon-user";
+IdentityProvider const IdentityProviderFacebook  = "oauth2-facebook";
+IdentityProvider const IdentityProviderApple     = "oauth2-apple";
+IdentityProvider const IdentityProviderUsernamePassword     = "local-userpass";
+
+std::string AppCredentials::serialize_as_json() const
+{
+    return m_payload_factory();
+}
+
+IdentityProvider provider_type_from_enum(AuthProvider provider)
```

Comment:
This doesn't seem to cover the full list? https://docs.mongodb.com/stitch/authentication/providers/

Reply:
I was intending to parallelise the rest of this work so that we can unblock.